src/lib/message/L1ToL2Message.ts

```diff
@@ -307,34 +308,36 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     // to do this we need to filter through the whole lifetime of the ticket looking
     // for relevant redeem scheduled events
     if (creationReceipt) {
-      const iFace = ArbRetryableTx__factory.createInterface()
-      const redeemTopic = iFace.getEventTopic('RedeemScheduled')
-
       let increment = 1000
       let fromBlock = await this.l2Provider.getBlock(
         creationReceipt.blockNumber
       )
       const creationBlock = await this.l2Provider.getBlock(
         creationReceipt.blockNumber
       )
+      let timeout = creationBlock.timestamp + l2Network.retryableLifetimeSeconds
+      const queriedRange: { from: number; to: number }[] = []
       const maxBlock = await this.l2Provider.getBlockNumber()
       while (fromBlock.number < maxBlock) {
         const toBlockNumber = Math.min(fromBlock.number + increment, maxBlock)
 
         // We can skip by doing fromBlock.number + 1 on the first go
         // since creationBlock because it is covered by the `getAutoRedeem` shortcut
-        const redeemEventLogs = await this.l2Provider.getLogs({
-          fromBlock: fromBlock.number + 1,
-          toBlock: toBlockNumber,
-          topics: [redeemTopic, this.retryableCreationId],
-        })
-        const redeemEvents = redeemEventLogs.map(
-          r => iFace.parseLog(r).args as RedeemScheduledEvent['args']
+        queriedRange.push({ from: fromBlock.number + 1, to: toBlockNumber })
```

Comment:
It's possible that the lifetime was extended in the first block, in which case I think we'd miss it.